avaje-config/src/main/java/io/avaje/config/ConfigParsers.java

```diff
@@ -3,12 +3,26 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Map.Entry;
 import java.util.Set;
 
-/**
- * Holds the non-properties ConfigParsers.
- */
-final class Parsers {
+public interface ConfigParsers {
+
+  /** Return the ConfigParser for the given extension. */
+  ConfigParser get(String extension);
+
+  /** Return true if the extension has a matching parser. */
+  boolean supportsExtension(String extension);
+
+  /** Return the set of supported extensions. */
+  Set<String> supportedExtensions();
+
+  /** Return the extension ConfigParser pairs. */
+  Set<Entry<String, ConfigParser>> entrySet();
+}
+
+/** Holds the ConfigParsers for various extension types. */
+final class Parsers implements ConfigParsers {
 
   private final Map<String, ConfigParser> parserMap = new HashMap<>();
```

Comment:
I have an uncomfortable feeling that this is not passed in the constructor. If it is not `ConcurrentHashMap` feels like the safer option here or possibly an immutable map depending on when `Parsers` is created.

Reply:
> this is not passed in the constructor

can you elaborate? what is this

Reply:
I mean it might as well be, nothing outside this class writes to it